trade_remedies_public/cases/views.py

```diff
@@ -379,6 +388,250 @@ def get(self, request, case_id, organisation_id=None, *args, **kwargs):
         )
 
 
+class InterestClientTypeStep2(LoginRequiredMixin, GroupRequiredMixin, FormMixin, BasePublicView):
+    groups_required = [SECURITY_GROUP_ORGANISATION_OWNER, SECURITY_GROUP_ORGANISATION_USER]
+    template_name = "v2/registration_of_interest/who_is_registering.html"
+    form_class = ClientTypeForm
+    case_page = True
+
+    def form_invalid(self, form):
+        form.assign_errors_to_request(self.request)
+        return super().form_invalid(form)
+
+    def get(self, request, case_id=None):
+        return render(
+            request,
+            self.template_name,
+            {
+                "case_id": case_id,
+            },
+        )
+
+    def post(self, request, case_id=None):
+        form = self.get_form()
+        if not form.is_valid():
+            return self.form_invalid(form)
+        elif form.cleaned_data.get("org") == "new-org":
+            return redirect(f"/case/interest/{case_id}/contact/")  # noqa: E501
+
+
+class InterestPrimaryContactStep2(
+    LoginRequiredMixin, GroupRequiredMixin, FormMixin, BasePublicView
+):
+    groups_required = [SECURITY_GROUP_ORGANISATION_OWNER, SECURITY_GROUP_ORGANISATION_USER]
+    template_name = "v2/registration_of_interest/primary_client_contact.html"
+    form_class = PrimaryContactForm
+    case_page = True
+
+    def form_invalid(self, form):
+        form.assign_errors_to_request(self.request)
+        return self.render_to_response(
```

Comment:
```suggestion
        return super().form_invalid(form=form)
```
And then override the get_context_data() method of the view to provide the case_id to the context